test/powershell/Language/Scripting/Requires.Tests.ps1

```diff
@@ -90,6 +90,46 @@ Describe "Requires tests" -Tags "CI" {
             { . $File } | Should -Throw -ExceptionType ([System.Management.Automation.ScriptRequiresException])
         }
     }
+
+    Context "Maxmimum PS Version checks" {
```

Comment:
You probably want to include a couple more test cases here:

- OS types specified with a bareword string
- Versions specified with a bareword string, like `6.2` and `6.2.3`
- We should also have a test defining the behaviour of `#requires -PSMaximumVersion 6`

Because this is a parsing change, you'll need to include negative tests too, where the `requires` does not parse properly. A few ideas for that:

- An OS type that is not a valid OS
- An OS type that is not a string
- A version that is a string but not a valid version

Reply:
Also, you should test the requires warning implementation using a test hook. It probably would look similar to:

https://github.com/PowerShell/PowerShell/blob/a96b4c040492e4e8f79ba243af6d43206fcd8791/test/powershell/engine/Help/HelpSystem.OnlineHelp.Tests.ps1#L10-L25

The test hook class is here:

https://github.com/PowerShell/PowerShell/blob/49e906bc3bd302e54ab1f5ad00c5bda8f8fba07f/src/System.Management.Automation/engine/Utils.cs#L2040-L2079

We just need to devise some way to communicate the warning that has been emitted. Possibly the answer is to just set the test hook value and check that.

The simple cases I'd imagine wanting to test are:

- No warning with no requires
- No warning with one requires
- No warning with many requires
- No warning when requires have blank lines between them as long as no other statements have been parsed
- Warning when a requires come after statements
- Warnings with different lines when multiple requires come after statements

Reply:
For an example of how to test parsing errors, see:

https://github.com/PowerShell/PowerShell/blob/49e906bc3bd302e54ab1f5ad00c5bda8f8fba07f/test/powershell/Language/Operators/PipelineChainOperator.Tests.ps1#L293-L302